### ISSUE TITLE
Use NetworkRequest for retrieving DNS servers

### ISSUE THREAD

**Issue by ameshkov** [AdguardTeam/AdguardForAndroid#1242]
https://developer.android.com/preview/behavior-changes.html

> * The system properties net.dns1, net.dns2, net.dns3, and net.dns4 are no longer available, a change that improves privacy on the platform.
> * To obtain networking information such as DNS servers, apps with the ACCESS_NETWORK_STATE permission can register a NetworkRequest or NetworkCallback object. These classes are available in Android 5.0 (API level 21) and higher.

**Comment by TPS** [comment]
I'm assuming this pertains only to those of us who _don't_ set custom DNS?

**Comment by ameshkov** [comment]
Yeah, just for the case when dns filtering is enabled and system default dns are in use

**Comment by ameshkov** [comment]
@Revertron aren't we already doing it in v2.11?

**Comment by Revertron** [comment]
No, for M+ we use `LinkProperties.getDnsServers()`. I think this is good.

**Comment by ameshkov** [comment]
But the original issue (no more `net.dns1`) is resolved ,right?

**Comment by Revertron** [comment]
Yep, we can close it.
